src/v/cloud_topics/level_zero/frontend_reader/reader.cc

```diff
@@ -445,4 +442,8 @@ bool level_zero_log_reader_impl::is_end_of_stream() const {
     return _current == state::end_of_stream_state;
 }
 
+bool level_zero_log_reader_impl::can_accept_batch(size_t size) const {
+    return (_config.strict_max_bytes || _config.bytes_consumed > 0)
+           && (_config.bytes_consumed + size) > _config.max_bytes;
```

Comment:
i realize this is the same logic as before, but something stood out:

it looks like we are checking if max bytes will be exceeded if the batch is accepted, rather than checking if max bytes has been exceeded after accepting at least one batch is accepted. does this have the (undesirable?) property that it might not make progress if the first batch alone exceeds the max?

Reply:
I just screwed up the naming. Let me fix

Reply:
Inverted the name so it reads correctly